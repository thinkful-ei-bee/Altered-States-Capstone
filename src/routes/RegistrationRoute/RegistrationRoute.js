import React, { Component } from "react";
import RegistrationForm from "../../components/RegistrationForm/RegistrationForm";
import happy from '../../images/happy.png'

class RegistrationRoute extends Component {
  state = {
    newUser: true
  }

  static defaultProps = {
    history: {
      push: () => {},
    }
  };

  handleRegistrationSuccess = () => {
    const { history } = this.props;
    history.push("/login");
  };

  render() {
    return (
      <section>
          {this.state.newUser && <div className='tutorial-container'>
            <div className='onboard-header'>
              <span>miMood</span>
              <span className='journal'>journal</span>
            </div>
            <img src={happy} alt='happy' />
            <h2>Welcome to miMood!</h2>
            <h3>Track your mood with the help of AI!</h3>
            <p>Machine learning algorithms analyze your entries for emotional cues, providing insight into how you're feeling.</p>
            <p>Add an entry to see emotional analysis.</p>
            <button onClick={() => this.setState({ newUser: false })}>GET STARTED</button>
          </div>}
        
        <RegistrationForm
          onRegistrationSuccess={this.handleRegistrationSuccess}
        />
      </section>
    );
  }
}

export default RegistrationRoute;
