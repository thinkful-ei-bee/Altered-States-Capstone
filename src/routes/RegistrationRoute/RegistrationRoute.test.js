import React from 'react';
import RegistrationRoute from './RegistrationRoute';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('DashboardRoute', () => {

  it('renders the complete page', () => {
    const wrapper = shallow(<RegistrationRoute />);
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})