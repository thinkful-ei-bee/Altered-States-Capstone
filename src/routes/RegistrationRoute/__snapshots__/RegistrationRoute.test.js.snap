// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRoute renders the complete page 1`] = `
<section>
  <RegistrationForm
    onRegistrationSuccess={[Function]}
  />
</section>
`;
